### PR TITLE
Add GCC 9

### PR DESCRIPTION
**Pull request by Croydon**
Changelog: Feature: Add GCC 9 containers in several flavors

- [x] I've read the [Contributing guide](https://github.com/conan-io/conan-docker-tools/blob/master/.github/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008) style guides for Python code.
- [x] I've followed the [Best Practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) guides for Dockerfile.

Bases on Ubuntu Disco

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -238,6 +238,41 @@ services:
...
+    gcc9:
+        build:
+            context: gcc_9
+            dockerfile: Dockerfile
+        image: "${DOCKER_USERNAME}/gcc9:${DOCKER_BUILD_TAG}"
+        container_name: gcc9
+        tty: true
+    gcc9-x96:
+        build:
+            context: gcc_9-x96
```
Comment (uilianries) [inline docker-compose.yml]:
```suggestion
            context: gcc_9-x86
```

File: docker-compose.yml [hunk 2]
```diff
@@ -238,6 +238,41 @@ services:
...
+            dockerfile: Dockerfile
+        image: "${DOCKER_USERNAME}/gcc9-x96:${DOCKER_BUILD_TAG}"
+        container_name: gcc9-x96
```
Comment (uilianries) [inline docker-compose.yml]:
```suggestion
        container_name: gcc9-x86
```

File: docker-compose.yml [hunk 3]
```diff
@@ -238,6 +238,41 @@ services:
...
+            context: gcc_9
+            dockerfile: Dockerfile
+        image: "${DOCKER_USERNAME}/gcc9:${DOCKER_BUILD_TAG}"
+        container_name: gcc9
+        tty: true
+    gcc9-x96:
+        build:
+            context: gcc_9-x96
+            dockerfile: Dockerfile
+        image: "${DOCKER_USERNAME}/gcc9-x96:${DOCKER_BUILD_TAG}"
```
Comment (uilianries) [inline docker-compose.yml]:
```suggestion
        image: "${DOCKER_USERNAME}/gcc9-x86:${DOCKER_BUILD_TAG}"
```

**Comment by Croydon** [comment]
Done

**Comment by Croydon** [comment]
Most tests fail due to https://github.com/conan-io/conan/pull/5046 not merged and in a release yet

I'm unfamiliar with the Jenkins Slave builds and why they fail. What's going on there?

**Comment by Croydon** [comment]
The base images GCC9 and GCC9_x86 build successful, the Jenkins Slave and ARM images fail. But since these build upon the base images, which won't get uploaded from a pull request, I believe that this pull request cannot succeed.

Since I haven't touch them Clang 7 and CentOS 6 fail unrelated

**Comment by Croydon** [comment]
I have rebased on the latest patches.

The problem with the Jenkins Slaves and ARM images persists. I don't think I can do anything to fix that.

For the master branch the problem of the ARM images could be solved with build stages, but not of pull requests runs like this ones, since without uploads the base images will still not be available despite build stage...


@uilianries Input please

**Comment by uilianries** [comment]
I'll provide multistage-build support ASAP. Also, I tried with Azure stages instead, but it didn't work as expected:

https://github.com/uilianries/conan-docker-tools/blob/feature/multiple-stage/azure-pipelines.yml

The second stage don't use generated images from first stage. No idea how to force it, but I sent a question to Azure forum.

**Comment by Croydon** [comment]
You are probably right, docker multi-stage builds seem to be a better, more universal solution to this problem 👍

**Comment by uilianries** [comment]
@Croydon could you please rebase your branch?

**Comment by Croydon** [comment]
Shouldn't #134 be merged first?

**Comment by uilianries** [comment]
well, you can update this one first, no problem. I can update #134 after :wink:

**Comment by Croydon** [comment]
Okay, but won't this pull request still fail then? 🤔

**Comment by uilianries** [comment]
oh, you are right, so let's wait #134

**Comment by lasote** [comment]
Merged the other PR

**Comment by lasote** [comment]
failing. Please check, ping me on slack for faster response

**Comment by uilianries** [comment]
I'm checking it right now

**Comment by lasote** [comment]
Still failing

**Comment by uilianries** [comment]
@lasote It's finally working! :green_heart:

**Pull request merged**